File: backend/internal/domain/thread.go
package domain

import "time"

// BookingThread is a private DM between the customer and the vendor of a single booking.
// Created automatically when the vendor accepts the booking.
type BookingThread struct {
	ID         string    `json:"id"`
	BookingID  string    `json:"bookingId"`
	CustomerID string    `json:"customerId"`
	VendorID   string    `json:"vendorId"`
	CreatedAt  time.Time `json:"createdAt"`
	UpdatedAt  time.Time `json:"updatedAt"`
}

// ThreadMessage is one message in a booking thread.
type ThreadMessage struct {
	ID        string    `json:"id"`
	ThreadID  string    `json:"threadId"`
	SenderID  string    `json:"senderId"`
	Text      string    `json:"text"`
	CreatedAt time.Time `json:"createdAt"`
}

File: mobile/lib/features/messaging/domain/entities/thread.dart
class BookingThread {
  const BookingThread({
    required this.id,
    required this.bookingId,
    required this.customerId,
    required this.vendorId,
    required this.createdAt,
    required this.updatedAt,
  });

  final String id;
  final String bookingId;
  final String customerId;
  final String vendorId;
  final String createdAt;
  final String updatedAt;
}

class ThreadMessage {
  const ThreadMessage({
    required this.id,
    required this.threadId,
    required this.senderId,
    required this.text,
    required this.createdAt,
  });

  final String id;
  final String threadId;
  final String senderId;
  final String text;
  final String createdAt;
}
